lib/fs.js

```diff
@@ -1836,30 +1836,42 @@ fs.FileWriteStream = fs.WriteStream; // support the legacy name
 
 
 WriteStream.prototype.open = function() {
+  const self = this;
+
   fs.open(this.path, this.flags, this.mode, function(er, fd) {
```

Comment:
They're permitted. Performance is on par w/ normal functions. Only place I'd suggest they be used is if `this` needs to propagate. Other than that it's nothing more than a style thing.